CONTRIBUTING.md

```diff
@@ -34,6 +34,14 @@ Signed commits are required. Follow [Github
 documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
 to set it up, and sign your commits so that your pull request can be merged.
 
+### Generated files
+
+Please make sure that generated files are checked in. This is done mainly to
+avoid broken builds and dependency issues. See #126 for more context.
+
+Mocks are generated in [mocks.go](./internal/pkg/mocks/mocks.go). They can be
+generated using `go generate ./...`.
+
```

Comment:
Do we need manually generate files each time that there is an update with `github.com/argoproj/argo-cd/v2/pkg/...`?
If yes. 

Can this cause merge conflicts? And do we know how often we need to regenerate?